kaggle-classification/trainer/eval.py

```diff
@@ -0,0 +1,93 @@
+"""
+Evaluates a model on data. Takes a path to a directory with a TensorFlow
+SavedModel and a path to a CSV with data, and outputs the predictions of the
+model on the data.
+
+Example usage:
+
+python trainer/eval.py  \
+  --model_dir=saved_models/{timestamp}/ \
+  --test_data=local_data/test.csv
+"""
+
+import argparse
+import numpy as np
+import pandas as pd
+import wikidata
+import tensorflow as tf
+
+# Name of the input words feature
+WORDS_FEATURE = 'words'
+
+# TODO: infer this from the saved model
+MAX_DOCUMENT_LENGTH = 500
+
+def predict(data, classifier):
+    """
+    Args:
+      data: n x m array of features where n is the number of examples
+            and m is the number of features.
+      classifier: a SavedModelPredictor with an input tensor 'input' that
+                  expects as input a list of tf.train.Example's.
+    Returns:
+      scores: a n x num_classes array of scores for each each class
+    """
+    model_inputs = []
+
+    for d in data:
```

Comment:
Let's change this to use the test_input_fn and classifier.predict(...) approach that we use in the model.py file.

Reply:
I looked into this, and it turns out that the `export_savedmodel` method on `Estimator` exports a `tensorflow.contrib.predictor.saved_model_predictor.SavedModelPredictor`, not an `Estimator`. Docs [here](https://www.tensorflow.org/api_docs/python/tf/contrib/learn/Estimator#export_savedmodel). So we can't use the same `classifier.predict()` approach to run data through the model. I couldn't figure out how to save an `Estimator` as an `Estimator`, but would love a pointer if you know how to do that because I agree it would be easier!